src/de/mod.rs

```diff
@@ -42,28 +42,32 @@ impl<'de> Deserializer<'de> {
     }
 
     pub fn from_str_with_options(input: &'de str, options: Options) -> SpannedResult<Self> {
-        Self::from_bytes_with_options(input.as_bytes(), options)
-    }
-
-    pub fn from_bytes_with_options(input: &'de [u8], options: Options) -> SpannedResult<Self> {
         let mut deserializer = Deserializer {
-            bytes: Bytes::new(input)?,
+            parser: Parser::new(input)?,
             newtype_variant: false,
             last_identifier: None,
             recursion_limit: options.recursion_limit,
         };
 
-        deserializer.bytes.exts |= options.default_extensions;
+        deserializer.parser.exts |= options.default_extensions;
 
         Ok(deserializer)
     }
 
+    pub fn from_bytes_with_options(input: &'de [u8], options: Options) -> SpannedResult<Self> {
+        Self::from_str_with_options(
+            str::from_utf8(input).map_err(|error| SpannedError::from_utf8_error(error, input))?,
+            options,
+        )
+    }
+
     pub fn remainder(&self) -> Cow<'_, str> {
-        String::from_utf8_lossy(self.bytes.bytes())
+        // FIXME this does not make sense with the unicode validation on creation
```

Comment:
While we could probably break the API here by always returning the remaining `&str`, it may be easier to give us more future wiggle room for now and just always return a borrowed `Cow` that just contains the remaining str